### PR TITLE
be sure that terminal target gets set for the instance

### LINKED ISSUE #138601: Background color is wrong when terminals are created in editor area via command
Repro is to run create new terminal in editor area command:

image

### LINKED ISSUE #138603: Terminal breaks when dragging to a different group, only when created in editor area via command
Repro is to run create new terminal in editor area command and drag and drop terminal to right group:

image

@lramos15 spotted this in https://github.com/microsoft/vscode/issues/138452#issuecomment-987181581

### LINKED ISSUE #138770: Terminal in Editor rendering errors and disposed errors
Does this issue occur when all extensions are disabled?: **Yes**

- VS Code Version: `1.63.0 (user setup)`
- OS Version: `Windows 10`

Steps to Reproduce:

1. Set the setting: `"terminal.integrated.defaultLocation": "editor",`
2. Open a new terminal (should open in editor).
   - Error Notification: `Terminal with ID 4 does not exist (has it already been disposed?)`
3. Open another new terminal in the editor.
   - Same message shows up
4. Drag split the terminal window to below.
   - content disappears from the bottom terminal, and is non interactive

I'm mostly repro-ing this with the WSL2 extension, but also happens with the standard windows powershell.

I also separately have VSCode Insiders installed, and have been seeing this since last month on the Insiders release. This issue has started happening in the standard release channel for me starting from yesterday's update.

This also happens on a fresh insiders install, and def with the wsl extension.

VS Code version: Code 1.63.0 (7db1a2b88f7557e0a43fec75b6ba7e50b3e9f77e, 2021-12-07T06:26:56.179Z)
OS version: Windows_NT x64 10.0.19043
Restricted Mode: No

### PR DESCRIPTION
**Pull request by meganrogge**

### CODE REVIEW

File: src/vs/workbench/contrib/terminal/browser/terminalInstance.ts [hunk 1]
```diff
@@ -635,6 +644,51 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 		return xterm;
 	}
 
+	private _getXtermTheme(theme?: IColorTheme): ITheme {
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/terminalInstance.ts:647]: I'd prefer this stayed in xtermTerminal, could we track target in TerminalInstance and pass it through to XtermTerminal in the ctor instead?

Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/terminalInstance.ts:647]: i'll try that

Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/terminalInstance.ts:647]: just looked into moving the tests that you have for this over to terminalInstance.test.ts and would be a pain, so this is a better idea

File: src/vs/workbench/contrib/terminal/browser/terminalInstance.ts [hunk 2]
```diff
@@ -179,12 +179,11 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
...
 	private _hasScrollBar?: boolean;
+	private _target?: TerminalLocation | undefined;
 
-	get target(): TerminalLocation | undefined { return this.xterm?.target; }
+	get target(): TerminalLocation | undefined { return this._target; }
 	set target(value: TerminalLocation | undefined) {
-		if (this.xterm) {
-			this.xterm.target = value;
-		}
+		this._target = value;
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/terminalInstance.ts:186]: We still need to be able to set the target after creation, let's also do `this.xterm.target = value` here and keep them in sync

**Review by Tyriar — CHANGES_REQUESTED** [review]

File: src/vs/workbench/contrib/terminal/browser/terminalInstance.ts [hunk 3]
```diff
@@ -179,12 +179,14 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 	private _labelComputer?: TerminalLabelComputer;
 	private _userHome?: string;
 	private _hasScrollBar?: boolean;
+	private _target?: TerminalLocation | undefined;
 
-	get target(): TerminalLocation | undefined { return this.xterm?.target; }
+	get target(): TerminalLocation | undefined { return this._target; }
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/terminalInstance.ts:184]: not sure why it won't let me do

```
this.xterm?.target = value 
```

File: src/vs/workbench/contrib/terminal/browser/terminalInstance.ts [hunk 4]
```diff
@@ -559,7 +561,7 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 			throw new Error('Terminal disposed of during xterm.js creation');
 		}
 
-		const xterm = this._instantiationService.createInstance(XtermTerminal, Terminal, this._configHelper, this._cols, this._rows);
+		const xterm = this._instantiationService.createInstance(XtermTerminal, Terminal, this._configHelper, this._cols, this._rows, this.target || TerminalLocation.Panel);
```
Comment (meganrogge) [inline src/vs/workbench/contrib/terminal/browser/terminalInstance.ts:564]: if target is undefined for the instance, should we throw or fallback to panel?

File: src/vs/workbench/contrib/terminal/browser/terminalInstance.ts (hunk 3 above)
Comment (Tyriar) [reply src/vs/workbench/contrib/terminal/browser/terminalInstance.ts:184]: That could end up resolving to `undefined = value` which isn't valid, afaik it's just not covered by that feature, need to do the if

**Review by Tyriar — APPROVED** [review]

File: test/smoke/src/areas/terminal/terminal-tabs.test.ts [hunk 5]
```diff
@@ -100,14 +100,14 @@ export function setup() {
 
 		it('should join tabs when more than one non-split terminal', async () => {
 			await terminal.runCommand(TerminalCommandId.Show);
-			await terminal.runCommand(TerminalCommandId.CreateNew);
+			await terminal.runCommandWithValue(TerminalCommandIdWithValue.CreateNew);
```
Comment (Tyriar) [inline test/smoke/src/areas/terminal/terminal-tabs.test.ts:103]: Shouldn't this remain runCommand as it doesn't take a value?

File: src/vs/workbench/contrib/terminal/browser/terminalInstance.ts (hunk 4 above)
Comment (Tyriar) [reply src/vs/workbench/contrib/terminal/browser/terminalInstance.ts:564]: Fallback seems fine

File: test/smoke/src/areas/terminal/terminal-tabs.test.ts (hunk 5 above)
Comment (meganrogge) [reply test/smoke/src/areas/terminal/terminal-tabs.test.ts:103]: it does take a value - "editor"

Comment (Tyriar) [reply test/smoke/src/areas/terminal/terminal-tabs.test.ts:103]: I don't see it getting passed in?

**Comment by Tyriar** [comment]
I reverted this as it broke the build 8605d7aeff46622320a70e577d0eba5c9b6557d9

**Pull request merged**